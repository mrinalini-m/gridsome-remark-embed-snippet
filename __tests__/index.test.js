const fs = require(`fs`),
  Remark = require(`remark`),
  GridsomeRemarkEmbedSnippet = require(`../index`),
  remark = new Remark()

function createPlugin(options = {}) {
  return GridsomeRemarkEmbedSnippet(options)
}

jest.mock(`fs`, () => {
  return {
    existsSync: jest.fn(),
    readFileSync: jest.fn()
  }
})

describe(`gridsome-remark-embed-snippet`, () => {
  beforeEach(() => {
    fs.existsSync.mockReset()
    fs.existsSync.mockReturnValue(true)
    fs.readFileSync.mockReset()
    fs.readFileSync.mockReturnValue(`console.log('testo');`)
  })

  test(`missing optional config options doesn't throw error`, () => {
    const mockMDContent = '`embed:test.js`',
      mockFile = { path: '/_tests_/test.md' },
      tree = remark.parse(mockMDContent),
      plugin = createPlugin({})

    plugin(tree, mockFile).then(results => {
      expect(results).toMatchSnapshot()
    })
  })

  test(`nonexistent specified directory throws error`, () => {
    fs.existsSync.mockReturnValue(false)

    const mockMDContent = '`embed:test.js`',
      mockFile = { path: '/_tests_/test.md' },
      tree = remark.parse(mockMDContent),
      plugin = createPlugin({ directory: `invalid` }),
      error = `Invalid directory specified "invalid"`

    return expect(plugin(tree, mockFile)).rejects.toThrow(error)
  })

  test(`invalid embed file path throws error`, () => {
    fs.existsSync.mockImplementation(path => path !== `examples/test.js`)

    const mockMDContent = '`embed:test.js`',
      mockFile = { path: '/_tests_/test.md' },
      tree = remark.parse(mockMDContent),
      plugin = createPlugin({ directory: `examples` }),
      error = `Invalid snippet specified; no such file "examples/test.js"`

    return expect(plugin(tree, mockFile)).rejects.toThrow(error)
  })
})

//Todo: Write more tests. Read more about tests.
// i_have_no_idea_what_i'm_doing.jpg
