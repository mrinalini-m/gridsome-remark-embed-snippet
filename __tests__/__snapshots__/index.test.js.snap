// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gridsome-remark-embed-snippet missing optional config options doesn't throw error 1`] = `undefined`;
